tests/results.cpp

```diff
@@ -1802,3 +1802,190 @@ TEST_CASE("results: snapshots") {
+    }
+
+    SECTION("Two integer properties combined") {
+        Results unique = results.distinct(SortDescriptor(results.get_tableview().get_parent(), {{0}, {2}}));
+        // unique is the same as the table
+        REQUIRE(unique.size() == N);
+        for (int i = 0; i < N; ++i) {
+            std::stringstream ss;
+            ss << "Foo_ " << i % 3;
+            REQUIRE(unique.get(i).get_string(1) == StringData(ss.str().c_str()));
+        }
+    }
+
+    SECTION("String and integer combined") {
+        Results unique = results.distinct(SortDescriptor(results.get_tableview().get_parent(), {{2}, {1}}));
+        // unique is the same as the table
+        REQUIRE(unique.size() == N);
+        for (int i = 0; i < N; ++i) {           
+            std::stringstream ss;
+            ss << "Foo_ " << i % 3;
+            REQUIRE(unique.get(i).get_string(1) == StringData(ss.str().c_str()));
+        }
+    }
+
+    // This section and next section demonstrate that sort().distinct() == distinct().sort()
+    SECTION("Order after sort and distinct") {
+        Results reverse = results.sort(SortDescriptor(results.get_tableview().get_parent(), {{2}}, {true}));
+        // reverse:
+        //   0, Foo_0,  1
+        //  ...
+        //   0, Foo_0, 10
+        REQUIRE(reverse.first()->get_int(2) == 1);
+        REQUIRE(reverse.last()->get_int(2) == 10);
+
+        // distinct() will first be applied to the table, and then sorting is reapplied
+        Results unique = reverse.distinct(SortDescriptor(reverse.get_tableview().get_parent(), {{0}}));
+        // unique:
+        //  2, Foo_2,  8
+        //  1, Foo_1,  9
+        //  0, Foo_0, 10
+        REQUIRE(unique.size() == 3);
+        REQUIRE(unique.get(0).get_int(2) == 8);
+        REQUIRE(unique.get(1).get_int(2) == 9);
+        REQUIRE(unique.get(2).get_int(2) == 10);
+    }
+
+    SECTION("Order after distinct and sort") {
+        Results unique = results.distinct(SortDescriptor(results.get_tableview().get_parent(), {{0}}));
+        // unique:
+        //  0, Foo_0, 10
+        //  1, Foo_1,  9
+        //  2, Foo_2,  8
+        REQUIRE(unique.size() == 3);
+        REQUIRE(unique.first()->get_int(2) == 10);
+        REQUIRE(unique.last()->get_int(2) == 8);
+        
+        // sort() is only applied to unique
+        Results reverse = unique.sort(SortDescriptor(unique.get_tableview().get_parent(), {{2}}, {true}));
+        // reversed:
+        //  2, Foo_2,  8
+        //  1, Foo_1,  9
+        //  0, Foo_0, 10
+        REQUIRE(reverse.size() == 3);
+        REQUIRE(reverse.get(0).get_int(2) == 8);
+        REQUIRE(reverse.get(1).get_int(2) == 9);
+        REQUIRE(reverse.get(2).get_int(2) == 10);
+    }
+
+    SECTION("Chaining distinct") {
+        Results first = results.distinct(SortDescriptor(results.get_tableview().get_parent(), {{0}}));
+        REQUIRE(first.size() == 3);
+        
+        // distinct() will discard the previous applied distinct() calls
```

Comment:
I tend to agree with you. 

Let's consider T = [(A, 1, alpha), (A, 2, beta), (B, 1, delta), (B, 2, gamma)]. 

V = T.distinct(0) = [(A, 1, alpha), (B, 1, delta)] and V.distinct(1) = [(A, 1, alpha)]
W = T.distinct(1) = [(A, 1, alpha), (A, 2, beta)] and W.distinct(0) = [(A, 1, alpha)]

On the other hand, T.distinct(0, 1) = [(A, 1, alpha), (A, 2, beta), (B, 1, delta), (B, 2, gamma)]

In short T.distinct(0).distinct(1) != T.distinct(0, 1) (I'm sure you can construct cases where equality is true).

This PR expose core's `distinct` method. From `table_view.hpp` in core:

```
// Each time you call distinct() it will first fetch the full original TableView contents and then apply
// distinct() on that. So it distinct() does not filter the result of the previous distinct().
```

The question is if we wish to change core's functionality? We might have good reasons for that but I believe that it is better to take that discussion in a core issue :-)